Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM ubuntu:latest
+MAINTAINER Shikher Verma "root@shikherverma.com"
+RUN apt-get update -y --fix-missing
+RUN apt-get install -y python python-pip python-dev build-essential git nodejs npm ruby ruby-dev
+# Install Dependencies
+RUN \
+  apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 7F0CEB10 && \
+  echo 'deb http://downloads-distro.mongodb.org/repo/ubuntu-upstart dist 10gen' > /etc/apt/sources.list.d/mongodb.list && \
+  apt-get update && \
+  apt-get install -y mongodb-org && \
+  rm -rf /var/lib/apt/lists/*
+COPY . /app
+WORKDIR /app
+RUN pip install -r requirements.txt
+RUN npm install --verbose
+RUN gem install sass
```

Comment:
We probably want to chain these with an *and* chain to reduce the number of layers.

Reply:
I also don't know if we want to use **four** different types of package managers, as they overlap their package offers and sometimes have different versions supported. I suggest that you:

1. use `apt's` [python-flask modules where possible.](https://packages.ubuntu.com/search?keywords=python-flask) Specially with such a simple requirements file.
2. use apt's gulp/other dependencies, and/or move the gulp part before it.
3. Use apt's bindings for libsass in python or whatever (which I assume is happening here). You are installing **three** more package managers for no reason at all (and again, it will result in a way bigger image). 

This whole thing also makes me wonder about the design of the whole stack, we can probably limit this to one/two langauges :P

Reply:
Re slimmer stack: Sure, what do you have in mind? I guess we can scrap Ruby and use a different sass implementation...
 
We'll probably have to keep Python (for back-end) and node (for front-end dependency management) unless you know a good Python-based front-end management system, or want to port the back-end to node? :P

Reply:
Basically `node/npm/gulp` and `ruby/sass` are only needed whenever the front-end dependencies or the `scss` code changes. 

During runtime we only need `Python/Flask/mod_wsgi` and `mongodb`.  A multi-stage build could separate these things.

Also note, that your dockerfile actually misses the front-end management calls, i.e.  a run of the `gulp`-script and the `scss` to `css` compilation.